### PR TITLE
Restore product image carousel

### PR DESCRIPTION
**Pull request by sacr3dc0w**
#### What?

Cornerstone lacks an image carousel for product images. This restores that functionality. Slick needs to be triggered on opening modal window for QuickView.

#### Screenshots

Gallery

### CODE REVIEW

**Comment by junedkazi** [comment]
@bigcommerce/stencil-team

File: templates/components/products/product-view.html [hunk 1]
```diff
@@ -160,7 +160,12 @@ <h2 class="modal-header-title">{{lang 'products.bulk_pricing.modal_title'}}</h2>
                      alt="{{product.main_image.alt}}" title="{{product.main_image.alt}}" data-main-image>
             </a>
         </figure>
-        <ul class="productView-thumbnails">
+        <ul class="productView-thumbnails" data-slick='{
+                "infinite": false,
+                "mobileFirst": true,
+                "slidesToShow": 4,
```
Comment (junedkazi) [inline templates/components/products/product-view.html]: Not sure we should limit to just 4 when we have the count from `product.images`

**Comment by sacr3dc0w** [comment]
updates
```
        <ul class="productView-thumbnails" data-slick='{
                "infinite": false,
                "mobileFirst": true,
                {{#gt product.images.length 6}}
                "slidesToShow": 6,
                {{else}}
                "slidesToShow": {{product.images.length}},
                {{/gt}}
                "slidesToScroll": 1
            }'>
```

After six thumbnails, things begin to get a little squished.

**Comment by junedkazi** [comment]
@sacr3dc0w how does it affect product pages which have more than 6 images. Lets say 10 images ? Can you provide a screenshot for that. Also how does it affect other screen sizes like tablet or mobile?

**Comment by sacr3dc0w** [comment]
@junedkazi Since the width for `.productView-thumbnail` is 20%, I've adjusted the logic to 5 images. The slick() method will only run if `product.images.length` is greater than 5.

For products that have more than 5 images, the `slick()` method is initialized and the additional images can be viewed by clicking next. The screenshots below that have both previous and next arrows visible are products where the next arrow was clicked to demo.

**Mobile**
mobile
**Tablet, landscape mode**
tablet, landscape
**Tablet, portrait mode**
tablet, portrait

**Comment by sacr3dc0w** [comment]
I decoded the Data URI SVG's for the `.slick-prev` and `.slick-next` classes and embedded them as inline SVG's.

- `$slick-arrow-prev`
- `$slick-arrow-next`

In doing so, we're able to fill the stroke of the SVG using `$slick-arrow-color`.

- `$slick-arrow-prev:              url("data:image/svg+xml;charset=utf8,%3Csvg xmlns='http://www.w3.org/2000/svg' width='24' height='42' viewBox='0 0 24 42'%3E%3Cpath d='M22.4572074 1.00746147l-21 20.02482143 20.9479397 19.9751786' stroke='#{$slick-arrow-color}' stroke-width='2' stroke-linecap='round' stroke-linejoin='round' fill='none' fill-rule='evenodd' /%3E%3C/svg%3E");`
- `$slick-arrow-next:              url("data:image/svg+xml;charset=utf8,%3Csvg xmlns='http://www.w3.org/2000/svg' width='24' height='42' viewBox='0 0 24 42'%3E%3Cpath d='M1.45679 1.00746147l21 20.02482143L1.50885 41.0074615' stroke='#{$slick-arrow-color}' stroke-width='2' stroke-linecap='round' stroke-linejoin='round' fill='none' fill-rule='evenodd' /%3E%3C/svg%3E");`

**Light**
1_light

**Bold**
2_bold

**Warm**
3_warm

4_theme-editor

**Pull request merged**